dtfs-central-api/src/services/utilisation-report-data-validator/utilisation-report-cell-validators/generate-ukef-facility-id-error.ts

```diff
@@ -0,0 +1,30 @@
+import { UKEF_FACILITY_ID_REGEX } from '../../../constants/regex';
+import { UtilisationReportCellValidationErrorGenerator } from './types/validation-error-generator';
+
+/**
+ * Generate and return error for the facility id cell if value is invalid
+ * @param facilityIdCellData - The cell data for the facility id cell
+ * @param exporterName - The name of the exporter for that csv row
+ * @returns The error if data is invalid, null if the data is valid
+ */
+export const generateUkefFacilityIdError: UtilisationReportCellValidationErrorGenerator = (facilityIdCellData, exporterName) => {
+  if (!facilityIdCellData?.value) {
+    return {
+      errorMessage: 'UKEF facility ID must have an entry',
+      column: facilityIdCellData?.column,
+      row: facilityIdCellData?.row,
+      value: facilityIdCellData?.value,
+      exporter: exporterName,
+    };
+  }
+  if (!UKEF_FACILITY_ID_REGEX.test(facilityIdCellData.value)) {
```

Comment:
I think this could be a helper.  Can pass it the REGEX, facilityIdCellData and exporterName and it returns the error for us

Reply:
We'd have to pass it quite a lot more information to make the error messaging accurate so I'm not sure the end outcome would be super valuable. Additionally, I think the amount of refactoring required would increase the scope of the ticket a fair amount which is to move the logic, not rework it extensively.

Reply:
I said similar @Zainzzkk https://github.com/UK-Export-Finance/dtfs2/pull/3413/files#r1740882018 , but I agree with @RLavender98 as this PR is about moving the validation to another place rather than code improvements. Would be nice to have a jira ticket to do in the future.

Reply:
Would be good to have a ticket for this